### DISCUSSION TITLE
Potential Slicer Feature Ideas

### DISCUSSION THREAD

**Discussion by AbeOwitz** [supermerill/SuperSlicer#1356] [Community contribution]
Just a couple of ideas I was thinking about recently...

1.  Has anyone tried ironing the first layer?  I was thinking we often print things upside down, but the first layer is sometimes ugly.  If we could iron the first layer, perhaps it would look better and stick to the bed better?  Would it reduce warping or reduce fade?

2.  Anyone looked into staggered perimeter heights?  Imagine a brick wall, with its staggered bricks.  Now rotate that wall 90 degrees.   

If we sliced the object perimeter vertically, we'd have something like this:
|-| |-|
| |-| |
|-| |-|
| |-| |

For example, each extrusion path is still 0.20 high, BUT each is always 0.10 offset from the adjacent ones.  Top and bottom layers are 1/2 the main layer height.

The slicer would put down 1/2 the layer at 0.10 high, then z+=0.20, and draw the other 1/2 at 0.20.  There would need to be 0.10 layers at the top and bottom to get the pattern started.

There would no longer be an uninterrupted layer strata across the object.  This could potentially improve layer adhesion, too.

**Comment by DzzD** [comment]
> 1. Has anyone tried ironing the first layer?  I was thinking we often print things upside down, but the first layer is sometimes ugly.  If we could iron the first layer, perhaps it would look better and stick to the bed better?  Would it reduce warping or reduce fade?

You can set ironing on first layer in "Print settings=>infill=>infill bottom : ironing

**Reply by AbeOwitz** [reply]
Is that the full post process ironing? What's the difference?  Going to try it anyway.  Thanks for pointing that out.

The ironing section header says this feature goes on top of inffill and perimeters, and gcode preview just shows regular infill.  

(I tend to get ripples in my first layer with large flat objects.  And each ripple perpetuates to the adjacent line.  If I microstep up, it does improve, but wont stick to the bed..  If I lower flow rate, it improves in some places.  Thought an ironing pass might smooth that out a bit.  )

**Reply by DzzD** [reply]
When using Ironing for filling bottom or top layer it make two passes, the second is 90° rotated and with twice as line (line are 50% thiner), you can see it clearly by zooming to the g-code preview, below you can see the first pass (in lite red) and the second pass red.

![image](https://user-images.githubusercontent.com/12198339/141760653-0b39699c-e518-4259-9cf8-8bc2ee528659.png)

**Comment by AbeOwitz** [comment]
Ohhh!!!!! I get it now.  I see it.  Sorry for not getting it!  

I just printed out the 1 layer 0.2 cube and it's ironing at the moment.  

Thanks for being patient and pointing that out.  :-)

**Comment by AbeOwitz** [comment]
On the other idea, I made some simple illustrations.  The theory is that interlacing/staggering lines in each layer so 1/2 is bound to the layer above, the other 1/2 below, we could make the object stronger.

I apologize in advance for my newb Blender skills...

![InterlacedLayers1](https://user-images.githubusercontent.com/44349159/141878941-875bb877-3e79-4618-84d4-bd497a89c89d.jpg)
![InterlacedLayers2](https://user-images.githubusercontent.com/44349159/141878942-66f5a962-0c8d-4022-bd99-2877e201ba5b.jpg)

How to do it?  Perhaps slicing at 1/2 the layer height, and alternating Z vs Z + 1/2LH every line spacing distance.  Not an easy thing to do, but it might improve strength.  Not all objects would need this, but based on CNC Kitchen and other testers, layer boundaries are the weakest direction.

Just an idea.  I've a lot more to learn before I tackle this.  But perhaps you all know of a better way...

**Comment by neophyl** [comment]
Except that the layer flow math normally puts enough plastic in that there are no gaps in the 'corners' at least on the inside of the part so strength wouldn't change.

**Comment by AbeOwitz** [comment]
That's just my newb Blender skills.  I realize there may or may not be gaps in the corners.  The drawing was designed to illustrate the difference and the idea in removing the horizontal strata as a weakness demonstrated in many filament tests.
